src/screens/Menu/AppSettings.js

```diff
@@ -0,0 +1,348 @@
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { FlatList } from 'react-native';
+import styled from 'styled-components/native';
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import { ScrollWrapper, Wrapper } from 'components/Layout';
+import SlideModal from 'components/Modals/SlideModal';
+import { spacing, fontStyles, fontTrackings } from 'utils/variables';
+import { supportedFiatCurrencies, defaultFiatCurrency } from 'constants/assetsConstants';
+import { MediumText, Paragraph, BaseText } from 'components/Typography';
+import SettingsListItem from 'components/ListItem/SettingsItem';
+import Button from 'components/Button';
+import Checkbox from 'components/Checkbox';
+import SystemInfoModal from 'components/SystemInfoModal';
+import {
+  saveBaseFiatCurrencyAction,
+  setAppThemeAction,
+  setUserJoinedBetaAction,
+  saveOptOutTrackingAction,
+} from 'actions/appSettingsActions';
+import { DARK_THEME, LIGHT_THEME } from 'constants/appSettingsConstants';
+
+
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
+
+import { SettingsSection } from './SettingsSection';
+
+type Props = {
+  baseFiatCurrency: ?string,
+  themeType: string,
+  userJoinedBeta: boolean,
+  optOutTracking: boolean,
+  saveBaseFiatCurrency: (currency: string) => void,
+  setAppTheme: (themeType: string, isManualThemeSelection?: boolean) => void,
+  setUserJoinedBeta: (status: boolean) => void,
+  saveOptOutTracking: (status: boolean) => void,
+};
+
+type State = {
+  visibleModal: ?string,
+  leaveBetaPressed: boolean,
+  joinBetaPressed: boolean,
+};
+
+const SettingsModalTitle = styled(MediumText)`
+  ${fontStyles.big};
+  margin: ${props => props.extraHorizontalSpacing ? `0 ${spacing.rhythm}px ${spacing.rhythm}px` : 0};
+`;
+
+const StyledWrapper = styled(Wrapper)`
+  justify-content: space-between;
+  padding-bottom: ${spacing.rhythm}px;
+  margin-top: ${spacing.medium}px;
+`;
+
+const SmallText = styled(BaseText)`
+  ${fontStyles.regular};
+  margin-top: 2px;
+  letter-spacing: ${fontTrackings.small}px;
+`;
+
+const CheckboxText = styled(BaseText)`
+  ${fontStyles.medium};
+  margin-top: 2px;
+  letter-spacing: ${fontTrackings.small}px;
+  margin-bottom: ${spacing.medium}px;
+`;
+
+
+const currencies = supportedFiatCurrencies.map(currency => ({ name: currency, value: currency }));
+
+class AppSettings extends React.Component<Props, State> {
+  state = {
+    visibleModal: null,
+    leaveBetaPressed: false,
+    joinBetaPressed: false,
+  }
+
+  renderListItem = (field: string, onSelect: Function, currentValue: string) => ({ item: { name, value } }: Object) => {
+    return (
+      <SettingsListItem
+        key={value}
+        label={name}
+        isSelected={value === currentValue}
+        onPress={() => onSelect({ [field]: value })}
+      />
+    );
+  };
```

Comment:
I think either splitting class method params per line or moving destruction part to new line return would make this cleaner. I'm not even sure how this went through lint's max line length 🤔